### ISSUE TITLE
Remontoire fails to load if i3/config is too large

### ISSUE THREAD

**Issue by buck10** [regolith-linux/regolith-desktop#383]
**Describe the bug**
There appears to be an issue with loading the remontoire help panel if the i3/config gets too large. The exact filesize where the issue occurs seems to vary depending on the number of lines present and amount of information on each line.

Note that the config file is loaded properly by i3 and all keybindings and settings seem to work correctly, it is just that the remontoire panel will not load via the mod+? keybinding or invocation of the remontoire-toggle script.

**To Reproduce**
One way to reproduce this behaviour (if a little unrealistically) is as follows:
- create a $HOME/regolith/i3/config file that contains 10914 lines with just a # on each line
- type i3-msg reload
- type /usr/bin/remontoire-toggle

Something like the following error should display in the terminal 
```
** (remontoire:13608): ERROR **: 21:55:29.559: main.vala:83: Failed to start: <data>:1:32754: Parse error: invalid string constant "#
#
#
#
#
#
#
#
#
#
#
#
#
#
#", expected valid string constant
```
with the journalctl log displaying something like:

`May 08 21:55:29 a******* kernel: traps: remontoire[13608] trap int3 ip:7f3a7d5d10d5 sp:7ffea24e4550 error:0 in libglib-2.0.so.0.6400.2[7f3a7d595000+84000]
`

Having just 10913 lines in the file will allow remontoire-toggle to execute properly and load the help panel (it will be empty).

I've attached the config file I was using when I found this issue.
[config.zip](https://github.com/regolith-linux/regolith-desktop/files/4602503/config.zip)

**Expected behavior**
Remontoire panel loads without error messages.

**Installation Details**
 - Regolith Install Type: both ISO and PPA
 - Regolith Version: 1.4
 - PPA url: ppa:regolith-linux/release
 - Host OS (for PPA): Ubuntu 20.04

**Additional context**
This seems like a fairly minor issue, but could be run into if the enhancement to allow configuration of the keybindings through the Xresources file is implemented.

I worked around the issue by modifying the remontoire-toggle script to call remontoire with the -c option like the following:

```
#!/bin/bash                                                                                                                                                 
# If remontoire is running, kill it.  Otherwise start it.

remontoire_PID=$(pidof remontoire)
local_config="$HOME/.config/regolith/i3/config"

if [ -z "$remontoire_PID" ]
then
  if [ -e "${local_config}" ]
  then
    /usr/bin/remontoire -c "${local_config}" &
  else
    /usr/bin/remontoire -s $(printenv I3SOCK) &
  fi  
else
    kill $remontoire_PID
fi

```

**Comment by moritzheiber** [comment]
Are there any drawbacks to your "fix"?

**Comment by kgilmer** [comment]
The issue seems to be a parse error, given `expected valid string constant`, yet that literal is not produced by remontoire.  @buck10 , the config attached is not 10K lines but 779.  Is there another config file that is longer?

**Comment by buck10** [comment]
Hi there, I've attached a file that reproduces the issue consistently. As mentioned originally, it has 10914 lines with a single '#' on each line. Replacing i3/config with this file and reloading the configuration via `i3-msg reload` and attempting to open the help with `remontoire-toggle` produces the error. Deleting a single line to get the file to 10913 lines removes the error case. Note that this isn't the only file that produces the result. Files with fewer lines, but more information per line (like a real config file) will also produce the result, but it is harder to pin point exactly where it breaks in those cases as it varies. It seems like it might be some sort of memory/buffer issue, but I'm just guessing as I'm not familiar enough with the sequence of what is getting executed.

Regarding the possible drawbacks of the workaround that I am using, it seems to me the major one is that there could be a disconnect between what configuration is actually loaded by Regolith and what Remontoire is reading from the config file. I think this is a fairly minor drawback, but it could be confusing to some users if the config file is edited, but not properly reloaded and Remontoire displays something other than what they expected. Also, the way I implemented the fix was a bit of a hack as I hard coded the reference to the config file which isn't really very flexible robust.

[config.errorcase.txt](https://github.com/regolith-linux/regolith-desktop/files/4671638/config.errorcase.txt)

**Comment by kgilmer** [comment]
Thank you @buck10 for providing the file for me to reproduce with.  I'll take a look and see what I find.

**Comment by kgilmer** [comment]
I've allocated a 32Kb buffer and that was ... just not enough.

**Comment by moritzheiber** [comment]
This has shipped with the latest release (1.5.1).
